GUI/Types/Renderer/GLTextureViewer.cs

```diff
@@ -781,19 +783,21 @@ private void UploadTexture(bool forceSoftwareDecode)
             }
 
             texture = GuiContext.MaterialLoader.LoadTexture(Resource, isViewerRequest: true);
-            decodeFlags = textureData.RetrieveCodecFromResourceEditInfo();
+            decodeFlags = textureData.RetrieveCodecFromResourceEditInfo() | swDecodeFlags;
         }
 
         private void UploadBitmap(SKBitmap bitmap)
         {
             Debug.Assert(bitmap != null);
-            Debug.Assert(bitmap.ColorType == SKColorType.Bgra8888);
 
             texture = new RenderTexture(TextureTarget.Texture2D, bitmap.Width, bitmap.Height, 1, 1);
-            decodeFlags = TextureCodec.None;
+            decodeFlags &= softwareDecodeOnlyOptions;
 
-            GL.TextureStorage2D(texture.Handle, 1, SizedInternalFormat.Rgba8, texture.Width, texture.Height);
-            GL.TextureSubImage2D(texture.Handle, 0, 0, 0, texture.Width, texture.Height, PixelFormat.Bgra, PixelType.UnsignedByte, bitmap.GetPixels());
+            var isHdr = bitmap.ColorType == SKColorType.RgbaF32;
```

Comment:
not textureData.IsHighDynamicRange?

Reply:
GenerateBitmap can return LDR bitmap with TextureCodec.ForceLdr

Reply:
This reminds me that the HW decoder does not respect that flag.